db/db.go

```diff
@@ -76,623 +70,746 @@ func New(db *sql.DB) DB {
-	) AS shifts WHERE tag0 = $1 ORDER BY created_time ASC LIMIT $2 OFFSET $3;`, hex.EncodeToString(tag[:]), pageSize, page*pageSize)
+// Txs implements the DB interface.
+func (db database) Txs(offset, limit int) ([]tx.Tx, error) {
+	txs := make([]tx.Tx, 0, limit)
+	rows, err := db.db.Query(`SELECT tableName, hash, selector, tx_id, amount, utxo_hash, utxo_index, value, pubkey_script, payload, phash, toAddr, nonce, nhash, gpubkey, ghash FROM (
+		SELECT 'lock_utxo_mint_account' AS tableName, hash, created_time, selector, '' AS tx_id, '' AS amount, utxo_hash, utxo_index, value, pubkey_script, payload, phash, toAddr, nonce, nhash, gpubkey, ghash FROM lock_utxo_mint_account UNION
+		SELECT 'lock_account_mint_account' AS tableName, hash, created_time, selector, tx_id, amount, '' AS utxo_hash, '' AS utxo_index, '' AS value, '' AS pubkey_script, payload, phash, toAddr, nonce, nhash, gpubkey, ghash FROM lock_account_mint_account UNION
+		SELECT 'burn_account_release_utxo' AS tableName, hash, created_time, selector, '' AS tx_id, amount, '' AS utxo_hash, '' AS utxo_index, '' AS value, '' AS pubkey_script, '' AS payload, '' AS phash, toAddr, nonce, '' AS nhash, '' AS gpubkey, '' AS ghash FROM burn_account_release_utxo UNION
+		SELECT 'burn_account_release_account' AS tableName, hash, created_time, selector, '' AS tx_id, amount, '' AS utxo_hash, '' AS utxo_index, '' AS value, '' AS pubkey_script, '' AS payload, '' AS phash, toAddr, nonce, '' AS nhash, '' AS gpubkey, '' AS ghash FROM burn_account_release_account
+	) AS shifts ORDER BY created_time ASC LIMIT $1 OFFSET $2;`, limit, offset)
 	if err != nil {
 		return nil, err
 	}
-	defer shifts.Close()
+	defer rows.Close()
 
-	// Loop through rows and convert them to txs.
-	for shifts.Next() {
-		tx, err := rowToShift(shifts)
+	// Loop through rows and convert them to transactions.
+	for rows.Next() {
+		tx, err := rowToTx(rows)
 		if err != nil {
 			return nil, err
 		}
 		txs = append(txs, tx)
 	}
-	return txs, shifts.Err()
+	return txs, rows.Err()
 }
 
-// PendingTxs implements the `DB` interface.
-func (db database) PendingTxs(expiry time.Duration) ([]abi.Tx, error) {
-	txs := make(abi.Txs, 0, 128)
-	shiftIns, err := db.db.Query(`SELECT hash, contract, p, token, toAddr, n, utxo_hash, utxo_vout FROM shift_in 
-	WHERE status = $1 AND $2 - created_time < $3;`, TxStatusConfirming, time.Now().Unix(), int64(expiry.Seconds()))
+// PendingTxs implements the DB interface.
+func (db database) PendingTxs(expiry time.Duration) ([]tx.Tx, error) {
+	txs := make([]tx.Tx, 0, 128)
+
+	// Get pending lock-and-mint (UTXO -> account) transactions from the database.
+	rows, err := db.db.Query(`SELECT hash, selector, utxo_hash, utxo_index, value, pubkey_script, payload, phash, toAddr, nonce, nhash, gpubkey, ghash FROM lock_utxo_mint_account
+		WHERE status = $1 AND $2 - created_time < $3;`, TxStatusConfirming, time.Now().Unix(), int64(expiry.Seconds()))
```

Comment:
Is there a reason we do not explicit order this query (and limit results to 128)? This would give consistent rows back.